.github/workflows/validate-query-formatting.yml

```
@@ -0,0 +1,42 @@
+name: "Validate Query Formatting"
+on:
+  merge_group:
+  pull_request:
+    branches:
+      - main
+      - next
+      - "rc/**"
+
+env:
+    XARGS_MAX_PROCS: 4
+
+jobs:
+  validate-query-formatting:
+      strategy:
+        matrix:
+          language: [cpp, c]
+      runs-on: ubuntu-latest
+      steps:
+        - name: Checkout
+          uses: actions/checkout@v3
+          with:
+            ref: ${{ inputs.ref }}
+
+        - name: Install CodeQL
+          run: |
+            VERSION="v$( jq -r '.supported_environment | .[0] | .codeql_cli' supported_codeql_configs.json)"
+            gh extensions install github/gh-codeql
+            gh codeql set-version "$VERSION"
+            gh codeql install-stub
+          env:
+            GITHUB_TOKEN: ${{ github.token }}
+
+        - name: Validate query format
+          env:
+            LANGUAGE: ${{ matrix.language }}
+          run: |
+            find $LANGUAGE -name \*.ql -or -name \*.qll -print0 | xargs -0 --max-procs "$XARGS_MAX_PROCS" codeql query format --in-place
```

I discovered recently that we need brackets around this, otherwise it only formats the `.qll` files. 😱 
```suggestion
            find $LANGUAGE \( -name \*.ql -or -name \*.qll \) -print0 | xargs -0 --max-procs "$XARGS_MAX_PROCS" codeql query format --in-place
```

---

Good catch.